lib/pengine/group.c

```diff
@@ -20,6 +20,33 @@
 #define VARIANT_GROUP 1
 #include "./variant.h"
 
+static bool
+skip_child_rsc(pe_resource_t *rsc, pe_resource_t *child, gboolean parent_passes,
+               GList *only_rsc, unsigned int show_opts)
+{
+    bool star_list = pcmk__list_of_1(only_rsc) &&
+                     pcmk__str_eq("*", g_list_first(only_rsc)->data, pcmk__str_none);
+    bool child_filtered = child->fns->is_filtered(child, only_rsc, FALSE);
+    bool child_active = child->fns->active(child, FALSE);
+    bool show_inactive = pcmk_is_set(show_opts, pcmk_show_inactive_rscs);
+
+    /* If the resource is in only_rsc by name (so, ignoring "*") then allow
+     * it regardless of if it's active or not.
+     */
+    if (!star_list && !child_filtered) {
+        return false;
+
+    } else if (!child_filtered && (child_active || show_inactive)) {
+        return false;
+
+    } else if (parent_passes && (child_active || show_inactive)) {
+        return false;
+
+    }
```

Comment:
FYI, is_filtered() is public API so we can't change it in any significant way, but glib guarantees that gboolean is an int, so we could change the last argument to int and make the value an enum (1/TRUE and 0/FALSE would have to keep their current behavior, but we could add e.g. 2 = the above tests with parent_passes==true and 3 = the above tests with parent_passes==false).

Not a recommendation, just letting you know what's possible with public APIs.